pkg/proxy/service/cluster_service_test.go

```diff
@@ -224,48 +278,102 @@ func (s *TestClusterServiceSuite) TestGetClusterAccess() {
 			},
 			func(si *service.ServiceImpl) {
 				si.GetMembersFunc = func(conditions ...commoncluster.Condition) []*commoncluster.CachedToolchainCluster {
-					return []*commoncluster.CachedToolchainCluster{
-						{
-							Config: &commoncluster.Config{
-								Name:        "member-1",
-								Type:        commoncluster.Member,
-								APIEndpoint: "https://api.endpoint.member-1.com:6443",
-								RestConfig: &rest.Config{
-									BearerToken: "def456",
-								},
-							},
-						},
-						{
-							Config: &commoncluster.Config{
-								Name:              "member-2",
-								APIEndpoint:       "https://api.endpoint.member-2.com:6443",
-								Type:              commoncluster.Member,
-								OperatorNamespace: "member-operator",
-								RestConfig: &rest.Config{
-									BearerToken: "abc123",
-								},
-							},
-							Client: memberClient,
-						},
+					return memberArray
+				}
+			},
+		)
+
+		s.Run("verify cluster access with route", func() {
+			memberClient.MockGet = func(ctx context.Context, key client.ObjectKey, obj client.Object) error {
+				route, ok := obj.(*routev1.Route)
+				if ok && key.Namespace == "tekton-results" && key.Name == "tekton-results" {
+					route.Namespace = key.Namespace
+					route.Name = key.Name
+					route.Status.Ingress = []routev1.RouteIngress{
 						{
-							Config: &commoncluster.Config{
-								Name:        "member-3",
-								APIEndpoint: "https://api.endpoint.member-3.com:6443",
-								Type:        commoncluster.Member,
-								RestConfig:  &rest.Config{},
-							},
+							Host: "https://api.endpoint.member-2.com:6443",
```

Comment:
This is the same URL as when the route is not set right? I think it'd be better to use a different URL for the route ingress for testing purposes. eg. `https://myservice.endpoint.member-2.com`